daemon/logger/jsonfilelog/read.go

```diff
@@ -305,20 +308,13 @@ func followLogs(f *os.File, logWatcher *logger.LogWatcher, notifyRotate chan int
 		if !since.IsZero() && msg.Timestamp.Before(since) {
 			continue
 		}
+		if !until.IsZero() && msg.Timestamp.After(until) {
+			continue
+		}
 		select {
 		case logWatcher.Msg <- msg:
 		case <-ctx.Done():
 			logWatcher.Msg <- msg
-			for {
```

Comment:
I wasn't exactly sure what this `for` block did, so I removed. It seems to just repeat what's already been done? If this is not the case and we need it, I can re-add.